docs/msi_v2/vm_vmss_credential_probe.md

```diff
@@ -0,0 +1,131 @@
+# Probe Logic for VM/VMSS Credential Endpoint
```

Comment:
Who is this file for? Won't we (MSAL) be directly attempting to contact the credential endpoint instead of "probing" it?

Reply:
I don't see any mention of a file @Robbie-Microsoft, MSAL needs to first determine all sources, so instead using the endpoint, we need to probe it, If we get a 400 then we use /credential endpoint, if not we fallback to IMDS

Reply:
I meant the file "[docs/msi_v2/vm_vmss_credential_probe.md](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/5096/files/f81b7b31a058ab2924f5f2dcdb1e7c7120a57f99#diff-97169d3ee04c84322eb0119a4e4e2e9c8ead763795203de511cd8f9c097d29f0)".

Regardless, we've never discussed this, and I don't see it in your end-to-end script.